scripts/run_e2e_tests.py

```diff
@@ -0,0 +1,440 @@
+
+
+def build_js_files(dev_mode):
+    """Build the javascript files.
+
+    Args:
+        dev_mode: boolean. Represents whether run the related commands in dev
+            mode.
+    """
+    tweak_constant_file(CONSTANT_FILE_PATH, dev_mode)
+    if not dev_mode:
+        python_utils.PRINT('  Generating files for production mode...')
+    else:
+        common.run_cmd(
+            [common.NODE_BIN_PATH, WEBPACK_BIN_PATH, '--config',
+             'webpack.dev.config.ts'])
+    build.main(args=(['--prod_env'] if not dev_mode else []))
+    os.remove('%s.bak' % CONSTANT_FILE_PATH)
+
+
+def tweak_webdriver_manager():
+    """webdriver-manager (version 13.0.0) uses `os.arch()` to determine the
+    architecture of the operation system, however, this function can only be
+    used to determine the architecture of the machine that compiled `node`
+    (great job!). In the case of Windows, we are using the portable version,
+    which was compiled on `ia32` machine so that is the value returned by this
+    `os.arch` function. While clearly the author of webdriver-manager never
+    considered windows would run on this architecture, so its own help function
+    will return null for this. This is causing the application has no idea
+    about where to download the correct version. So we need to change the
+    lines in webdriver-manager to explicitly tell the architecture.
+
+    https://github.com/angular/webdriver-manager/blob/b7539a5a3897a8a76abae7245f0de8175718b142/lib/provider/chromedriver.ts#L16
+    https://github.com/angular/webdriver-manager/blob/b7539a5a3897a8a76abae7245f0de8175718b142/lib/provider/geckodriver.ts#L21
+    https://github.com/angular/webdriver-manager/blob/b7539a5a3897a8a76abae7245f0de8175718b142/lib/provider/chromedriver.ts#L167
+    https://github.com/nodejs/node/issues/17036
+    """
+    regex_pattern = PATTERN_FOR_REPLACE_WEBDRIVER_CODE
+    arch = 'x64' if common.is_x64_architecture() else 'x86'
+    replace = 'this.osArch = "%s";' % arch
+    common.inplace_replace_file(
+        CHROME_PROVIDER_FILE_PATH, regex_pattern, replace)
+    common.inplace_replace_file(
+        GECKO_PROVIDER_FILE_PATH, regex_pattern, replace)
+
+
+def undo_webdriver_tweak():
+    """Undo the tweak on webdriver manager's source code."""
+    if os.path.isfile(CHROME_PROVIDER_BAK_FILE_PATH):
+        os.remove(CHROME_PROVIDER_FILE_PATH)
+        os.rename(CHROME_PROVIDER_BAK_FILE_PATH, CHROME_PROVIDER_FILE_PATH)
+    if os.path.isfile(GECKO_PROVIDER_BAK_FILE_PATH):
+        os.remove(GECKO_PROVIDER_FILE_PATH)
+        os.rename(GECKO_PROVIDER_BAK_FILE_PATH, GECKO_PROVIDER_FILE_PATH)
+
+
+def start_webdriver_manager():
+    """Update and start webdriver manager."""
+    if common.is_windows_os():
+        tweak_webdriver_manager()
+
+    run_webdriver_manager(
+        ['update', '--versions.chrome', CHROME_DRIVER_VERSION])
+    run_webdriver_manager(
+        ['start', '--versions.chrome', CHROME_DRIVER_VERSION,
```

Comment:
There were two `start` on the `bash` script, but it looks like the second one is not working. I tried to run the two `start` manually and the second one was throwing an error that the port 4444 was already bound.